### PR TITLE
Dpu t-tabularToRelational support for xls and DBF

### PR DESCRIPTION
**Pull request by eea02**
We need support for XLS/X and DBF files in tabularToRelational DPU.

### CODE REVIEW

File: t-tabularToRelational/pom.xml [hunk 1]
```diff
@@ -1,63 +1,82 @@
...
-		<artifactId>uv-pom-dpu</artifactId>
-		<version>2.1.0-SNAPSHOT</version>
-		<relativePath/>
-	</parent>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>eu.unifiedviews</groupId>
+        <artifactId>uv-pom-dpu</artifactId>
+        <version>2.1.0-SNAPSHOT</version>
```
Comment (skrchnavy) [inline t-tabularToRelational/pom.xml]: uv-pom-dpu version 2.1.0 released so shall be used.

File: t-tabularToRelational/pom.xml [hunk 2]
```diff
@@ -1,63 +1,82 @@
...
-	<artifactId>uv-t-tabularToRelational</artifactId>
-	<name>T-TabularToRelational</name>
-	<description>Parse tabular file to relational data unit.</description>
-	<version>2.1.0-SNAPSHOT</version>
-	<packaging>bundle</packaging>
+    <groupId>eu.unifiedviews.plugins</groupId>
+    <artifactId>uv-t-tabularToRelational</artifactId>
+    <name>T-TabularToRelational</name>
+    <description>Parse tabular file to relational data unit.</description>
+    <version>2.1.0-SNAPSHOT</version>
```
Comment (skrchnavy) [inline t-tabularToRelational/pom.xml]: https://github.com/UnifiedViews/Plugins/blob/master/t-tabularToRelational/pom.xml#L16 v22/1/0 version released shall be 2.2.0-SNAPSHOT

File: t-tabularToRelational/README.md [hunk 3]
```diff
@@ -38,7 +40,7 @@
 
 |Version |Release notes |
 |--------|--------------|
-|2.1.0   | improved user experience; Update to API 2.1.0 |
+|2.1.0   | improved user experience; Update to API 2.1.0; added new parsers(XLS, XLSX, DBF) |
```
Comment (skrchnavy) [inline t-tabularToRelational/README.md]: new version 2.2.0 to e used for new features.

File: t-tabularToRelational/pom.xml [hunk 4]
```diff
@@ -1,63 +1,82 @@
...
+            <artifactId>dbf-reader</artifactId>
+            <version>0.1.0</version>
+        </dependency>
+        <!-- UNIT Test dependencies -->
+        <dependency>
+            <groupId>eu.unifiedviews</groupId>
+            <artifactId>module-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (skrchnavy) [inline t-tabularToRelational/pom.xml]: this is in uv-pom-dpu so duplicated/

File: t-tabularToRelational/pom.xml [hunk 5]
```diff
@@ -1,63 +1,82 @@
...
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>junit</groupId>
-			<artifactId>junit</artifactId>
-			<scope>test</scope>
-		</dependency>
-	</dependencies>
+    <dependencies>
+        <dependency>
```
Comment (skrchnavy) [inline t-tabularToRelational/pom.xml]: is in uv-pom-dpu so not need to duplicate

**Comment by eea02** [comment]
Made changes according to comments from @skrchnavy.

**Comment by skrchnavy** [comment]
Due to change in configuration (need to support cfg. migration) this is not planed in 2.1.1 but in later release.

**Comment by mkl-eea-sk** [comment]
and what about integrating new POI deps

**Comment by tomas-knap** [comment]
Ok, approved (Note: I did not tested the functionality)

Note: This DPU and t-tabular should have most of the logic quite similar, the difference is in the output - in one case, it is relational table, in second case it is RDF. Is there any alignment between t-tabular and t-tabularToRelational?

**Comment by eea02** [comment]
There are some differences, for example headers from input files are ignored and must be provided explicitly by the user. This is because it is easier for user to make operations with relational data unit, when he knows exact name of columns.
For parsing CSV, tabularToRelational uses bulk data load of H2 database. For XLS/X and DBF it uses the same libraries as t-tabular

**Comment by skrchnavy** [comment]
Is migration of configuration properly tested? Is it 100% backward compatible.

**Pull request merged**